packages/babel-plugin-react-i18n/src/test/BabelPluginReactI18n.test.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from 'path';
+import {transformFile} from '@babel/core';
+import BabelPluginReactI18n from '../BabelPluginReactI18n';
+
+function runFixture(fixturePath: string) {
+  return new Promise<string>((resolve, reject) => {
+    transformFile(
+      fixturePath,
+      {
+        plugins: [BabelPluginReactI18n],
+        sourceRoot: __dirname,
+        filename: fixturePath,
+        filenameRelative: path.relative(__dirname, fixturePath),
+      },
+      (err, result) => {
+        if (err) {
+          reject(err);
+        } else {
+          // eslint-disable-next-line typescript/no-non-null-assertion
+          resolve(result!.code!);
+        }
+      },
+    );
+  });
+}
+
+describe('babel-pluin-react-i18n', () => {
+  it('injects arguments when adjacent translations exist', async () => {
+    const fixturePath = path.resolve(
+      __dirname,
+      'fixtures',
+      'adjacentTranslations',
+      'MyComponent.js',
+    );
+
+    const code = await runFixture(fixturePath);
+    expect(code).toMatchInlineSnapshot(`
+"import React from 'react';
```

Comment:
This is a nitpick: I did snapshots in webgen to start but have since written a custom matcher that calls a few helpers for comparing file contents (stips their indents or formats, etc..). Historically we don't like snapshots and there isn't any reason this should be an exception IMO. I'd suggest using the matcher I wrote or writing a simple helper function. Then re-write these tests to be more specific (that it adds the import, generated an id, transformed the arguments, etc...). This might be overkill for such a small package/transformed code, but just thought I'd point it out.

Reply:
I do think this makes sense overall, but I agree it may be overkill for such a small package.